### ISSUE TITLE
Runtime Issue with Ruby Client on aarch64-linux

### ISSUE THREAD

**Issue by marocchino** [Unleash/yggdrasil#173]
### Describe the bug

I have identified an error occurring during try to process in certain environments. Could you please look into this?
I have confirmed that this issue does not occur on arm base macOS.

### Steps to reproduce the bug

```
pksys@pkprotodev2046:~$ gem install unleash
Fetching ffi-1.16.3.gem
Fetching yggdrasil-engine-0.0.7-aarch64-linux.gem
Fetching unleash-6.0.7.gem
Building native extensions. This could take a while...
Successfully installed ffi-1.16.3
Successfully installed yggdrasil-engine-0.0.7-aarch64-linux
Successfully installed unleash-6.0.7
Parsing documentation for ffi-1.16.3
Installing ri documentation for ffi-1.16.3
Parsing documentation for yggdrasil-engine-0.0.7-aarch64-linux
Installing ri documentation for yggdrasil-engine-0.0.7-aarch64-linux
Parsing documentation for unleash-6.0.7
Installing ri documentation for unleash-6.0.7
Done installing documentation for ffi, yggdrasil-engine, unleash after 5 seconds
3 gems installed

A new release of RubyGems is available: 3.5.16 → 3.5.23!
Run `gem update --system 3.5.23` to update your installation.

pksys@pkprotodev2046:~$ irb
irb(main):001> require 'unleash'
/usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/ffi-1.16.3/lib/ffi/library.rb:216:in `attach_function': Function 'new_engine' not found in [/lib/aarch64-linux-gnu/libc.so.6] (FFI::NotFoundError)
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/yggdrasil-engine-0.0.7-aarch64-linux/lib/yggdrasil_engine.rb:52:in `<class:YggdrasilEngine>'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/yggdrasil-engine-0.0.7-aarch64-linux/lib/yggdrasil_engine.rb:48:in `<top (required)>'
	from <internal:/usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
	from <internal:/usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/unleash-6.0.7/lib/unleash/toggle_fetcher.rb:5:in `<top (required)>'
	from <internal:/usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
	from <internal:/usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/unleash-6.0.7/lib/unleash/client.rb:2:in `<top (required)>'
	from <internal:/usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
	from <internal:/usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/unleash-6.0.7/lib/unleash.rb:6:in `<top (required)>'
	from <internal:/usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:141:in `require'
	from <internal:/usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:141:in `rescue in require'
	from <internal:/usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:135:in `require'
	from (irb):1:in `<main>'
	from <internal:kernel>:187:in `loop'
	... 3 levels...
<internal:/usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require': cannot load such file -- unleash (LoadError)
	from <internal:/usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
	from (irb):1:in `<main>'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/irb/workspace.rb:121:in `eval'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/irb/workspace.rb:121:in `evaluate'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/irb/context.rb:633:in `evaluate_expression'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/irb/context.rb:600:in `evaluate'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/irb.rb:1049:in `block (2 levels) in eval_input'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/irb.rb:1380:in `signal_status'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/irb.rb:1041:in `block in eval_input'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/irb.rb:1120:in `block in each_top_level_statement'
	from <internal:kernel>:187:in `loop'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/irb.rb:1117:in `each_top_level_statement'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/irb.rb:1040:in `eval_input'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/irb.rb:1021:in `block in run'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/irb.rb:1020:in `catch'
	from /usr/dena/home/pksys/.rbenv/versions/3.3.5/lib/ruby/3.3.0/irb.rb:1020:in `run'
	... 4 levels...
```

### Expected behavior

```
$ irb
irb(main):001> require 'unleash'
=> true
```

### Logs, error output, etc.

```bash
$ ruby -v
ruby 3.2.5 (2024-07-26 revision 31d0f1a2e7) [aarch64-linux]
$ bundle list
Gems included by the bundle:
  * CFPropertyList (2.3.6)
  * actioncable (7.1.4.1)
  * actionmailbox (7.1.4.1)
  * actionmailer (7.1.4.1)
  * actionpack (7.1.4.1)
  * actiontext (7.1.4.1)
  * actionview (7.1.4.1)
  * active_decorator (1.4.1)
  * active_flag (2.0.2)
  * active_hash (3.3.1)
  * activejob (7.1.4.1)
  * activejob-cancel (0.3.2)
  * activemodel (7.1.4.1)
  * activerecord (7.1.4.1)
  * activerecord-cause (0.7.0)
  * activerecord-import (1.8.1)
  * activerecord-refresh_connection (0.0.5)
  * activestorage (7.1.4.1)
  * activesupport (7.1.4.1)
  * addressable (2.8.7)
  * aes_key_wrap (1.1.0)
  * amazing_print (1.6.0)
  * anbt-sql-formatter (0.1.0)
  * apple_id (1.6.4)
  * asset_finder (2.2.0)
  * ast (2.4.2)
  * attr_required (1.0.1)
  * autodoc (0.8.0)
  * autoprefixer-rails (10.4.16.0)
  * aws-eventstream (1.3.0)
  * aws-partitions (1.994.0)
  * aws-sdk-batch (1.104.0)
  * aws-sdk-core (3.211.0)
  * aws-sdk-ivs (1.63.0)
  * aws-sdk-ivsrealtime (1.34.0)
  * aws-sdk-kms (1.95.0)
  * aws-sdk-lambda (1.139.0)
  * aws-sdk-rekognition (1.110.0)
  * aws-sdk-s3 (1.170.0)
  * aws-sdk-sns (1.90.0)
  * aws-sdk-sts (1.11.0)
  * aws-sigv4 (1.9.1)
  * base64 (0.2.0)
  * bcrypt (3.1.20)
  * benchmark_driver (0.16.5)
  * better_errors (2.10.1)
  * bigdecimal (3.1.8)
  * bindata (2.4.15)
  * bindex (0.8.1)
  * binding_of_caller (1.0.1)
  * bootsnap (1.18.4)
  * bootstrap-sass (3.4.1)
  * boxr (1.21.1)
  * builder (3.2.4)
  * bullet (7.2.0)
  * byebug (11.1.3)
  * capybara (3.40.0)
  * caxlsx (4.1.0)
  * choice (0.2.0)
  * chronic (0.10.2)
  * coderay (1.1.3)
  * coercible (1.0.0)
  * coffee-rails (5.0.0)
  * coffee-script (2.4.1)
  * coffee-script-source (1.12.2)
  * committee (5.4.0)
  * committee-rails (0.8.0)
  * concurrent-ruby (1.3.4)
  * config (5.5.2)
  * connection_pool (2.4.1)
  * counter_culture (3.8.2)
  * crass (1.0.6)
  * daizu (1.6.0)
  * daizu-error (2.22.0)
  * daizu-global_infra_logger (0.1.1)
  * daizu-header (0.7.0)
  * daizu-locale (1.4.0)
  * daizu-ope (1.23.3)
  * dalli (3.2.8)
  * dalli-elasticache (1.0.1)
  * database_rewinder (1.0.1)
  * datadog (2.6.0)
  * datadog-ruby_core_source (3.3.6)
  * date (3.3.4)
  * db-query-matchers (0.13.0)
  * debug_inspector (1.2.0)
  * declarative (0.0.20)
  * deep_merge (1.2.2)
  * descendants_tracker (0.0.4)
  * diff-lcs (1.5.1)
  * diffy (3.4.2)
  * digest-crc (0.6.5)
  * docile (1.4.0)
  * dogstatsd-ruby (5.6.3)
  * domain_name (0.6.20240107)
  * doorkeeper (5.7.1)
  * doorkeeper-openid_connect (1.8.9)
  * dragonfly (1.4.0)
  * dragonfly-s3_data_store (1.3.0)
  * drb (2.2.1)
  * elastic-transport (8.3.2)
  * elasticsearch (8.14.0)
  * elasticsearch-api (8.14.0)
  * elasticsearch-model (8.0.0)
  * elasticsearch-persistence (8.0.0)
  * elasticsearch-rails (8.0.0)
  * email_validator (2.2.4)
  * enumerize (2.7.0)
  * erubi (1.12.0)
  * et-orbi (1.2.11)
  * exception_notification (4.5.0)
  * excon (0.109.0)
  * execjs (2.7.0)
  * factory_bot (6.5.0)
  * factory_bot_rails (6.4.4)
  * faraday (2.12.0)
  * faraday-follow_redirects (0.3.0)
  * faraday-multipart (1.0.4)
  * faraday-net_http (3.3.0)
  * faraday-retry (2.2.0)
  * ffaker (2.23.0)
  * ffi (1.16.3)
  * ffi-compiler (1.0.1)
  * fog-aws (3.21.0)
  * fog-core (2.4.0)
  * fog-json (1.2.0)
  * fog-xml (0.1.4)
  * formatador (1.1.0)
  * fugit (1.11.1)
  * gapic-common (0.21.1)
  * gems (1.2.0)
  * get_process_mem (0.2.7)
  * git (2.3.1)
  * globalid (1.2.1)
  * god (0.13.7)
  * gon (6.4.0)
  * google-api-client (0.53.0)
  * google-apis-bigquery_v2 (0.78.0)
  * google-apis-core (0.15.1)
  * google-apis-discovery_v1 (0.14.0)
  * google-apis-drive_v3 (0.59.0)
  * google-apis-generator (0.12.0)
  * google-apis-iamcredentials_v1 (0.22.0)
  * google-apis-storage_v1 (0.46.0)
  * google-cloud-bigquery (1.50.0)
  * google-cloud-core (1.6.1)
  * google-cloud-env (2.2.0)
  * google-cloud-errors (1.4.0)
  * google-cloud-storage (1.52.0)
  * google-cloud-vision (1.5.1)
  * google-cloud-vision-v1 (0.15.0)
  * google-cloud-vision-v1p3beta1 (0.13.1)
  * google-id-token (1.4.2)
  * google-protobuf (3.25.5)
  * google_spreadsheet_fetcher (2.1.0)
  * googleapis-common-protos (1.4.0)
  * googleapis-common-protos-types (1.11.0)
  * googleauth (1.11.2)
  * grpc (1.67.0)
  * grpc-tools (1.67.0)
  * guard (2.18.1)
  * guard-compat (1.2.1)
  * guard-rspec (4.7.3)
  * hashids (1.0.6)
  * hashie (5.0.0)
  * htmlentities (4.3.4)
  * http (5.1.1)
  * http-cookie (1.0.5)
  * http-form_data (2.3.0)
  * httpclient (2.8.3)
  * i18n (1.14.4)
  * io-console (0.7.2)
  * irb (1.14.0)
  * itunes_receipt_decoder (0.3.1)
  * itunes_receipt_validator (0.0.7)
  * jb (0.8.2)
  * jbuilder (2.13.0)
  * jmespath (1.6.2)
  * jquery-rails (4.6.0)
  * json (2.7.2)
  * json-jwt (1.16.6)
  * json_schema (0.21.0)
  * jsonpath (1.1.5)
  * jwt (2.9.3)
  * kakeibo (0.0.2)
  * kaminari (1.2.2)
  * kaminari-actionview (1.2.2)
  * kaminari-activerecord (1.2.2)
  * kaminari-core (1.2.2)
  * kgio (2.11.4)
  * koala (3.6.0)
  * language_server-protocol (3.17.0.3)
  * libdatadog (14.0.0.1.0)
  * libddwaf (1.15.0.0.0)
  * listen (3.9.0)
  * llhttp-ffi (0.4.0)
  * log4r (1.1.10)
  * logger (1.6.0)
  * loofah (2.22.0)
  * lumberjack (1.2.10)
  * m3u8 (0.8.2)
  * mail (2.8.1)
  * marcel (1.0.2)
  * matrix (0.4.2)
  * method_source (1.0.0)
  * mime-types (3.5.2)
  * mime-types-data (3.2023.1205)
  * mini_mime (1.1.5)
  * minitest (5.22.3)
  * mongon (1.4.0 4eb8b56)
  * msgpack (1.7.2)
  * multi_db_migration (0.2.0)
  * multi_json (1.15.0)
  * multi_xml (0.6.0)
  * multipart-post (2.3.0)
  * mutex_m (0.2.0)
  * mysql2 (0.5.6)
  * nenv (0.3.0)
  * net-http (0.4.1)
  * net-imap (0.4.10)
  * net-ldap (0.19.0)
  * net-pop (0.1.2)
  * net-protocol (0.2.2)
  * net-smtp (0.5.0)
  * nio4r (2.7.1)
  * nokogiri (1.16.7)
  * notiffany (0.1.3)
  * oauth (0.6.2)
  * oauth2 (2.0.9)
  * octoball (0.1.6.1)
  * octokit (9.2.0)
  * omniauth (2.1.2)
  * omniauth-apple (1.3.0)
  * omniauth-facebook (10.0.0)
  * omniauth-google-oauth2 (1.2.0)
  * omniauth-line (0.1.0)
  * omniauth-oauth (1.2.0)
  * omniauth-oauth2 (1.8.0)
  * omniauth-twitter (1.4.0 0f9ad01)
  * oneline_log_formatter (0.0.2)
  * openapi_parser (2.1.0)
  * openid_connect (2.3.0)
  * os (1.1.4)
  * ostruct (0.6.0)
  * parallel (1.26.3)
  * parallel_tests (4.7.2)
  * parser (3.3.1.0)
  * process_executer (1.1.1)
  * pry (0.14.2)
  * pry-byebug (3.10.1)
  * pry-doc (1.5.0)
  * pry-rails (0.3.11)
  * pry-remote (0.1.8)
  * pry-stack_explorer (0.6.1)
  * psych (5.1.2)
  * public_suffix (6.0.1)
  * puma (6.4.3)
  * raabro (1.4.0)
  * racc (1.8.1)
  * rack (2.2.9)
  * rack-cors (2.0.2)
  * rack-health (0.1.2)
  * rack-oauth2 (2.2.1)
  * rack-protection (3.2.0)
  * rack-server_status (0.0.4)
  * rack-session (1.0.2)
  * rack-test (2.1.0)
  * rack-user_agent (0.5.3)
  * rackup (1.0.0)
  * rails (7.1.4.1)
  * rails-dom-testing (2.2.0)
  * rails-erd (1.7.2)
  * rails-html-sanitizer (1.6.0)
  * rails_autolink (1.1.8)
  * railties (7.1.4.1)
  * rainbow (3.1.1)
  * raindrops (0.20.1)
  * rake (13.2.1)
  * rb-fsevent (0.11.2)
  * rb-inotify (0.10.1)
  * rchardet (1.8.0)
  * rdoc (6.7.0)
  * redcarpet (3.6.0)
  * redis (4.8.1)
  * redis-client (0.22.2)
  * redis-namespace (1.11.0 ddcc4b9)
  * redis-session-store (0.11.5)
  * regexp_parser (2.9.0)
  * regexp_trie (1.0.3)
  * reline (0.5.9)
  * representable (3.2.0)
  * request_store (1.7.0)
  * request_store-sidekiq (0.1.0)
  * retriable (3.1.2)
  * rexml (3.3.9)
  * ridgepole (2.1.1)
  * roo (2.10.1)
  * rouge (4.2.1)
  * rspec (3.13.0)
  * rspec-core (3.13.0)
  * rspec-expectations (3.13.0)
  * rspec-mocks (3.13.0)
  * rspec-rails (7.0.1)
  * rspec-support (3.13.1)
  * rspec_junit_formatter (0.6.0)
  * rubocop (1.68.0)
  * rubocop-ast (1.32.3)
  * rubocop-capybara (2.21.0)
  * rubocop-factory_bot (2.26.1)
  * rubocop-performance (1.22.1)
  * rubocop-rails (2.27.0)
  * rubocop-rspec (3.2.0)
  * rubocop-rspec_rails (2.30.0)
  * ruby-graphviz (1.2.5)
  * ruby-progressbar (1.13.0)
  * ruby-ulid (0.9.0)
  * rubyzip (2.3.2)
  * sass-rails (6.0.0)
  * sassc (2.4.0)
  * sassc-rails (2.1.2)
  * sawyer (0.9.2)
  * shellany (0.0.1)
  * sidekiq (7.3.2)
  * sidekiq-cron (1.12.0)
  * signet (0.19.0)
  * silencer (2.0.0)
  * simple_oauth (0.3.1)
  * simple_twitter (2.2.2)
  * simplecov (0.22.0)
  * simplecov-html (0.12.3)
  * simplecov_json_formatter (0.1.4)
  * slack-notifier (2.4.0)
  * slim (5.2.0)
  * slim-rails (3.6.3)
  * slop (3.6.0)
  * snaky_hash (2.0.1)
  * spring (4.2.1)
  * spring-commands-rspec (1.0.4)
  * spring-commands-thor (1.0.1)
  * spring-watcher-listen (2.1.0)
  * sprockets (3.7.2)
  * sprockets-rails (3.4.2)
  * sti_type_customizable (0.5.0)
  * stringio (3.1.1)
  * swd (2.0.3)
  * symmetric-encryption (4.6.0)
  * temple (0.10.3)
  * tencentcloud-sdk-common (3.0.937)
  * tencentcloud-sdk-trtc (3.0.937)
  * terminal-notifier (2.0.0)
  * terminal-notifier-guard (1.7.0)
  * terser (1.2.4)
  * thor (1.3.1)
  * thread_safe (0.3.6)
  * tilt (2.3.0)
  * timeout (0.4.1)
  * trailblazer-option (0.1.2)
  * tzinfo (2.0.6)
  * uber (0.1.0)
  * unicode-display_width (2.5.0)
  * unicorn (6.1.0)
  * unicorn-worker-killer (0.4.5)
  * uniform_notifier (1.16.0)
  * unleash (6.0.7)
  * uri (0.13.1)
  * validate_url (1.0.15)
  * version_gem (1.1.4)
  * warning (1.4.0)
  * weak_parameters (0.5.0)
  * web-console (4.2.1)
  * webfinger (2.1.2)
  * webrick (1.8.2)
  * websocket-driver (0.7.6)
  * websocket-extensions (0.1.5)
  * whenever (1.0.0)
  * woothee (1.13.0)
  * worker_scoreboard (0.0.5)
  * xpath (3.2.0)
  * yaml_vault (1.3.2)
  * yard (0.9.36)
  * yggdrasil-engine (0.0.7)
  * zeitwerk (2.6.13)
  * zip-codes (0.3.3)
Use `bundle info` to print more detailed information about a gem
$ bundle platform
Your platform is: aarch64-linux
...
$ rbenv exec bundle exec thor secret_task:decrypt --only-current-rails-env=true
bundler: failed to load command: thor (/data/pksys/tmp/provisioning/bundle/ruby/3.2.0/bin/thor)
/data/pksys/tmp/provisioning/bundle/ruby/3.2.0/gems/ffi-1.16.3/lib/ffi/library.rb:216:in `attach_function': Function 'new_engine' not found in [/lib/aarch64-linux-gnu/libc.so.6] (FFI::NotFoundError)
	from /data/pksys/tmp/provisioning/bundle/ruby/3.2.0/gems/yggdrasil-engine-0.0.7-aarch64-linux/lib/yggdrasil_engine.rb:52:in `<class:YggdrasilEngine>'
	from /data/pksys/tmp/provisioning/bundle/ruby/3.2.0/gems/yggdrasil-engine-0.0.7-aarch64-linux/lib/yggdrasil_engine.rb:48:in `<main>'
...
$ ls -al /data/pksys/tmp/provisioning/bundle/ruby/3.2.0/gems/yggdrasil-engine-0.0.7-aarch64-linux/lib/
total 1432
drwxrwxr-x 2 pksys pksys        4096 Nov 12 13:20 .
drwxr-xr-x 4 pksys infrawork    4096 Nov 12 13:20 ..
-rw-r--r-- 1 pksys pksys        1697 Nov 12 13:20 custom_strategy.rb
-rw-r--r-- 1 pksys pksys     1441992 Nov 12 13:20 libyggdrasilffi_aarch64.so
-rw-r--r-- 1 pksys pksys        4203 Nov 12 13:20 yggdrasil_engine.rb
```

### Subscription type

Pro

### Hosting type

Hosted by Unleash

### SDK information (language and version)

ruby 6.0.7

**Comment by sighphyre** [comment]
Oh, sorry about that, I think this was fixed in the beta range of 6.0.8. Let me run some tests to confirm and I'll do a release. Apologies!

**Comment by sighphyre** [comment]
This should be fixed in the 6.0.8 version of the Unleash client
